src/ToEnum.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter;
+
+use BackedEnum;
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Stdlib\ArrayUtils;
+use Traversable;
+use UnitEnum;
+
+use function is_array;
+use function is_int;
+use function is_string;
+use function is_subclass_of;
+
+/**
+ * @psalm-type Options array{enum: class-string<UnitEnum>}
+ * @extends AbstractFilter<Options>
+ */
+class ToEnum extends AbstractFilter
```

Comment:
I think this should be made more generic:

```php
/**
 * @template TEnum of UnitEnum
 */
```

Then in the constructor you can:

```php
/**
 * @param class-string<TEnum> $enum
 */
public function __construct(string $enum)
{
    // ...
}
```

This way, `filter()` becomes generic, and you can infer the correct ENUM type too:

```php
/**
 * @return TEnum|null
 */
public function filter(mixed $value): ?UnitEnum
{
    // ...
}
```

Reply:
I tried it but psalm will not understand my type
`psalm: UndefinedDocblockClass: Docblock-defined class, interface or enum named Laminas\Filter\TEnum does not exist`
when I use `class-string<TEnum>|Traversable|Options`. If I remove the `Traversable|Options` psalm understands it.